src/Lean/Util/Trace.lean

```diff
@@ -210,6 +236,12 @@ macro "trace[" id:ident "]" s:(interpolatedStr(term) <|> term) : doElem => do
     if (← Lean.isTracingEnabledFor cls) then
       Lean.addTrace cls $msg)
 
+macro "with_trace[" id:ident "]" s:(interpolatedStr(term) <|> termBeforeDo) " do " cont:doSeq : doElem => do
```

Comment:
I'm not sure if this macro is worth it, it's used only three times in total and doesn't even support the fancier (and preferable) `withTraceNode*` combinators.

And unlike `trace[...]`, it computes `$msg` even if the trace class is disabled.

Reply:
Ah, a remnant of my original plan to simply upgrade the `trace` macro, which turned out not to be the best idea